### PR TITLE
ABU-700, ABU-706, ABU-701, ABU-703, ABU-704, ABU-705, ABU-796

### PR DESCRIPTION
**Pull request by oliverfoster**
* ABU-704 - Have change assessment _postScoreToLms to _includeInTotalScore

* ABU-705 - Added _requireAssessmentPassed to total assessment config.

* ABU-706 - techfix: Have changed so that the running model represents all defaults and configurations at runtime, allowing any change to the model to be reflected in the assessments behaviours

* ABU-796 -  The assessment will now reset the question states and progress on "the user literally navigating away from the web-page" (unless the spoor extended tracking is enabled).
With spoor extended tracking and no _isResetOnRevisit on the assessment, it would remember the question selections and the assessment progress.

* ABU-703 - techfix: The main file has been renamed to 'assessment.js' as with 'offlineStorage' in adapt core. note: This file should probably be in the framework rather than the contrib-assessment

* ABU-701 - On page revisit, checks if the assessment is complete, checks if the assessment is in a second session and check if the questions have not been restored, then reset the assessment
If requireAssessmentPassed: true, set attempts to 'infinite' to stop the user being unable to complete the course.
Remember assessment states on menus so that pages need not be revisited once passed on a second session.

Additional:

* Have update readme  
* Have moved getConfig in assessmentModel to make it part of the global api spec.

Should be checked with:
https://github.com/adaptlearning/adapt_framework/pull/659
https://github.com/adaptlearning/adapt-contrib-pageLevelProgress/pull/54
https://github.com/adaptlearning/adapt-contrib-boxmenu/pull/47

### CODE REVIEW

File: js/adapt-assessmentArticleModel.js [hunk 1]
```diff
@@ -439,6 +439,31 @@ define([
             });
         },
 
+        _checkIfQuestionWereRestored: function() {
```
Comment (moloko) [inline js/adapt-assessmentArticleModel.js]: should be ```_checkIfQuestionsWereRestored```?

**Comment by dancgray** [comment]
+1

**Comment by hrajotia** [comment]
+1

**Comment by zenduo** [comment]
+1

**Pull request merged**